src/base/net/proxyconfigurationmanager.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2016  Vladimir Golovnev <glassez@yandex.ru>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "proxyconfigurationmanager.h"
+#include "base/settingsstorage.h"
+
+#define SETTINGS_KEY(name) "Network/Proxy/" name
+const QString KEY_DISABLED = SETTINGS_KEY("Disabled");
+const QString KEY_TYPE = SETTINGS_KEY("Type");
+const QString KEY_IP = SETTINGS_KEY("IP");
+const QString KEY_PORT = SETTINGS_KEY("Port");
+const QString KEY_AUTHENTICATION = SETTINGS_KEY("Authentication");
+const QString KEY_USERNAME = SETTINGS_KEY("Username");
+const QString KEY_PASSWORD = SETTINGS_KEY("Password");
+
+namespace
+{
+    inline SettingsStorage *settings() { return  SettingsStorage::instance(); }
+
+    inline bool isSameConfig(const Net::ProxyConfiguration &conf1, const Net::ProxyConfiguration &conf2)
```

Comment:
Comparing configurations outside this module isn't needed (we can know about changes via ProxyConfiguration signal) so I don't want to have `operator==()` (and even other comparing methods) in ProxyConfiguration interface.